src/ircode.c

```diff
@@ -843,16 +842,12 @@ JL_DLLEXPORT jl_string_t *jl_compress_ir(jl_method_t *m, jl_code_info_t *code)
     // by the various jl_ir_ accessors. Make sure to adjust those if you change
     // the data layout.
 
-    for (i = 0; i < 6; i++) {
+    for (i = 0; i < 5; i++) {
         int copy = 1;
         if (i == 1) { // skip codelocs
             assert(jl_field_offset(jl_code_info_type, i) == offsetof(jl_code_info_t, codelocs));
             continue;
         }
-        if (i == 4) { // don't copy contents of method_for_inference_limit_heuristics field
-            assert(jl_field_offset(jl_code_info_type, i) == offsetof(jl_code_info_t, method_for_inference_limit_heuristics));
-            copy = 0;
-        }
```

Comment:
This seems still relevant, even though the index has changed from 4 to 8, the content didn't change

Reply:
Field 8 is not serialized. The field, like the other generated function inputs does not get compressed.

Reply:
Where does this get preserved then if a generation function passes to the CodeInstance?

Reply:
We never compress the output of a generated function.